python/ray/data/block.py

```diff
@@ -458,3 +458,25 @@ def aggregate_combined_blocks(
     def block_type(self) -> BlockType:
         """Return the block type of this block."""
         raise NotImplementedError
+
+
+def _get_block_boundaries(
```

Comment:
```suggestion
def _get_grouping_keys_boundaries(
```

Reply:
@wingkitlee0 apologies for the initial confusion that my comments caused! 

I've misread the intent of your changes and hence some of my comments weren't actually relevant for what you're trying to do.